instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/extensions/bedrock.py

```diff
@@ -255,29 +271,88 @@ def _extract_nova_attributes(self, attributes, request_body):
             attributes, GEN_AI_REQUEST_MAX_TOKENS, config.get("max_new_tokens")
         )
         self._set_if_not_none(
-            attributes,
-            GEN_AI_REQUEST_STOP_SEQUENCES,
-            config.get("stopSequences"),
+            attributes, GEN_AI_REQUEST_STOP_SEQUENCES, config.get("stopSequences")
         )
 
     def _extract_claude_attributes(self, attributes, request_body):
         self._set_if_not_none(
-            attributes,
-            GEN_AI_REQUEST_MAX_TOKENS,
-            request_body.get("max_tokens"),
+            attributes, GEN_AI_REQUEST_MAX_TOKENS, request_body.get("max_tokens")
         )
         self._set_if_not_none(
-            attributes,
-            GEN_AI_REQUEST_TEMPERATURE,
-            request_body.get("temperature"),
+            attributes, GEN_AI_REQUEST_TEMPERATURE, request_body.get("temperature")
         )
         self._set_if_not_none(
             attributes, GEN_AI_REQUEST_TOP_P, request_body.get("top_p")
         )
         self._set_if_not_none(
-            attributes,
-            GEN_AI_REQUEST_STOP_SEQUENCES,
-            request_body.get("stop_sequences"),
+            attributes, GEN_AI_REQUEST_STOP_SEQUENCES, request_body.get("stop_sequences")
+        )
+
+    def _extract_command_r_attributes(self, attributes, request_body):
+        prompt = request_body.get("message")
+        self._set_if_not_none(
+            attributes, GEN_AI_USAGE_INPUT_TOKENS, math.ceil(len(prompt) / 6)
```

Comment:
Do you have a reference on why we need this division for cohere and mistral?

Reply:
Cohere and Mistral models don't provide input and output token counts in response body, so we impute these values from the input and output lengths. Bedrock uses 6 chars per token to approximate token count for pricing, per https://docs.aws.amazon.com/bedrock/latest/userguide/model-customization-prepare.html. Provided this context in comment.

Reply:
Thanks